sys/kern/pci.c

```diff
@@ -31,22 +31,41 @@ static const pci_vendor_id *pci_find_vendor(uint16_t vendor_id) {
 }
 
 static bool pci_device_present(device_t *pcid) {
-  return pci_read_config(pcid, PCIR_DEVICEID, 4) != -1U;
+  return pci_read_config_4(pcid, PCIR_DEVICEID) != -1U;
 }
 
 static int pci_device_nfunctions(device_t *pcid) {
+  /* If first function of a device is invalid, then
+   * no more functions are present. */
   if (!pci_device_present(pcid))
     return 0;
-  uint32_t hdrtype = pci_read_config(pcid, PCIR_HEADERTYPE, 1);
+  uint32_t hdrtype = pci_read_config_1(pcid, PCIR_HEADERTYPE);
```

Comment:
`uint8_t hdrtype` to match `pci_read_config_1` ?

Reply:
Done.